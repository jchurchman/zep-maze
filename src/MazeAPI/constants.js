export const FETCHING_MAZES = 'FETCHING_MAZES';
export const FETCHING_MAZE = 'FETCHING_MAZE';
export const FETCHED_MAZES = 'FETCHED_MAZES';
export const FETCHED_MAZE = 'FETCHED_MAZE';
export const FETCHED_MAZE_ERROR = 'FETCHED_MAZE_ERROR';
export const ADDING_MAZE = 'ADDING_MAZE';
export const ADDED_MAZE = 'ADDED_MAZE';
export const ADDING_MAZE_ERROR = 'ADDING_MAZE_ERROR';