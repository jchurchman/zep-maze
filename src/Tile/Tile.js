import React from 'react';
import styled from 'styled-components';

const Square = styled.td`
    height: 16vw;
    width: 16vw;
    box-sizing: border-box;
    border: 6px outset #a9a9a9;
    text-align: center;
    vertical-align: middle;
`

export function Tile({ value = '' }) {
    return (
        <Square className={`${value.toLowerCase()}`}>
        </Square>
    );
}