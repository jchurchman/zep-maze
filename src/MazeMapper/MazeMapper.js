import React, { Component } from 'react';
import { Tile } from '../Tile/Tile';
import styles from './MazeMapper.css';

export default class MazeMapper extends Component{

    render() {
        const { position, maze } = this.props;


        const xVal = (position[0] - 1) * -48;
        const yVal = (position[1] - 1) * -48;
        
        const transform = `translate( ${ xVal }px, ${ yVal }px )`;

        return (
            <table className={styles.maze} style={{transform}}>
                <tbody>
                    {maze.matrix.map( ( row, index ) => {
                        return (
                            <tr key={index}>
                            {row.map( ( tile, index ) => {
                                return <Tile key={index} value={tile}/>
                            })}
                            </tr>
                        )
                    })}
                </tbody>
            </table>
    )}
}