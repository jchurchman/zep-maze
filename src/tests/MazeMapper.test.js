import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import MazeMapper from '../MazeMapper/MazeMapper';
import { WALL, FLOOR, START, END } from '../Tile/Tile.constants';

const maze = {
    score: [],
    matrix: [
        [ WALL, WALL, WALL, WALL, WALL ],
        [ WALL, START, WALL, END, WALL ],
        [ WALL, FLOOR, WALL, FLOOR, WALL ],
        [ WALL, FLOOR, FLOOR, FLOOR, WALL ],
        [ WALL, WALL, WALL, WALL, WALL ],
    ]
}

describe('MazeMapper component', () => {

    it('renders a maze table when given a maze object', () => {
        const wrapper = shallow(<MazeMapper maze={maze} position={[ 1, 1 ]}/>);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });
    
})