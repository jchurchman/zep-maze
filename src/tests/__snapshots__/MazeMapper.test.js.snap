// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MazeMapper component renders a maze table when given a maze object 1`] = `
<styled.table
  style={
    Object {
      "transform": "translate( 0vw, 0vw )",
    }
  }
>
  <tbody>
    <tr>
      <Tile
        value="WALL"
      />
      <Tile
        value="WALL"
      />
      <Tile
        value="WALL"
      />
      <Tile
        value="WALL"
      />
      <Tile
        value="WALL"
      />
    </tr>
    <tr>
      <Tile
        value="WALL"
      />
      <Tile
        value="START"
      />
      <Tile
        value="WALL"
      />
      <Tile
        value="END"
      />
      <Tile
        value="WALL"
      />
    </tr>
    <tr>
      <Tile
        value="WALL"
      />
      <Tile
        value="FLOOR"
      />
      <Tile
        value="WALL"
      />
      <Tile
        value="FLOOR"
      />
      <Tile
        value="WALL"
      />
    </tr>
    <tr>
      <Tile
        value="WALL"
      />
      <Tile
        value="FLOOR"
      />
      <Tile
        value="FLOOR"
      />
      <Tile
        value="FLOOR"
      />
      <Tile
        value="WALL"
      />
    </tr>
    <tr>
      <Tile
        value="WALL"
      />
      <Tile
        value="WALL"
      />
      <Tile
        value="WALL"
      />
      <Tile
        value="WALL"
      />
      <Tile
        value="WALL"
      />
    </tr>
  </tbody>
</styled.table>
`;
